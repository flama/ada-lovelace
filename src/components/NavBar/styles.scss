@import '../../config';

#navbar {
  @include wrap-mobile;

  > .left, > .right { padding: $space-xxxs; }
  > .right { margin-left: $space-xxs; }

  @include medium-only {
    @include grid(6);
    @include offsetLeft(1);
  }

  @include large-up {
    @include grid(10);
    @include offsetLeft(1);
    @include offsetRight(2);
  }

  height: 15vh;

  > .left, > .right {
    margin-top: $space-s;

    @include medium-up {
      margin-top: $space-xl;
    }
    padding: $space-xxs;
    font-size: $font-size-centi;
    font-weight: bold;
    text-transform: uppercase;
    letter-spacing: 1px;
    color: $light-purple;
  }

  > .left {
    display: block;
    float: left;
  }

  > .right {
    display: block;
    float: right;
    margin-left: $space-xs;
  }

  > * { transition: all 0.3s ease; }

  > *:hover { color: #fff; }
}
