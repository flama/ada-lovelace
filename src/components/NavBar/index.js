import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-scroll'

import './styles.scss'

const NavBar = props => {
  return (
    <div id="navbar">
      <a href="http://flama.is" target="_blank" className="left">
        made with â™¥ by flama
      </a>
      <div className="right">
        <div className={ `nav-burger${props.open? ' -open':''}` }
          onClick={ () => props.toggleMenu() }>
        </div>
        <div className="nav-items">
          <Link to="share"
            smooth="easeInOutQuad"
            duration={500}
            className="right">
            share
          </Link>
          <Link to="about-page"
            smooth="easeInOutQuad"
            duration={500}
            className="right">
            about us
          </Link>
        </div>
      </div>
    </div>
  )
}

NavBar.propTypes = {
  open: PropTypes.bool,
  toggleMenu: PropTypes.func,
}

export default NavBar
