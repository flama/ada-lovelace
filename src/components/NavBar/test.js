import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import NavBar from '../NavBar'

describe('the NavBar', () => {
  let navbar

  beforeEach(() => {
    navbar = shallow(<NavBar />)
  })

  it('renders correctly', () => {
    expect(toJSON(navbar)).toMatchSnapshot()
  })

  it('adds .-open to .nav-burger when open', () => {
    navbar.setProps({ open: true })
    expect(navbar.find('.nav-burger').hasClass('-open')).toBeTruthy()
  })

  it('calls toggleMenu prop when nav-burger is clicked', () => {
    let toggleMenu = jest.fn()
    navbar.setProps({ toggleMenu })
    navbar.find('.nav-burger').simulate('click')
    expect(toggleMenu).toHaveBeenCalled()
  })
})
