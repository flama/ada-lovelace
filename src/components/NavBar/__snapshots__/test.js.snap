// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the NavBar renders correctly 1`] = `
<div
  id="navbar"
>
  <a
    className="left"
    href="http://flama.is"
    target="_blank"
  >
    made with â™¥ by flama
  </a>
  <div
    className="right"
  >
    <div
      className="nav-burger"
      onClick={[Function]}
    />
    <div
      className="nav-items"
    >
      <a
        className="right"
        onClick={[Function]}
      >
        share
      </a>
      <a
        className="right"
        onClick={[Function]}
      >
        about us
      </a>
    </div>
  </div>
</div>
`;
